server/src/main/java/org/elasticsearch/action/synonyms/GetSynonymsAction.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.synonyms;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.synonyms.SynonymRule;
+import org.elasticsearch.synonyms.SynonymsManagementAPIService;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+public class GetSynonymsAction extends ActionType<GetSynonymsAction.Response> {
+
+    public static final GetSynonymsAction INSTANCE = new GetSynonymsAction();
+    public static final String NAME = "cluster:admin/synonyms/get";
+
+    public GetSynonymsAction() {
+        super(NAME, Response::new);
+    }
+
+    public static class Request extends ActionRequest {
+        private static final int MAX_SEARCH_RESULTS = 10_000;
+        private final String synonymsSetId;
+        private final int from;
+        private final int size;
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.synonymsSetId = in.readString();
+            this.from = in.readVInt();
+            this.size = in.readVInt();
+        }
+
+        public Request(String SynonymsSetId, int from, int size) {
+            Objects.requireNonNull(SynonymsSetId, "Synonym set ID cannot be null");
+            this.synonymsSetId = SynonymsSetId;
+            this.from = from;
+            this.size = size;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            ActionRequestValidationException validationException = null;
+
+            validationException = validatePageParam("from", from, validationException);
+            validationException = validatePageParam("size", size, validationException);
+
+            return validationException;
+        }
+
+        private ActionRequestValidationException validatePageParam(
+            String paramName,
+            int value,
+            ActionRequestValidationException validationException
+        ) {
+            if (value < 0) {
+                validationException = addValidationError("[" + paramName + "] must be a positive integer", validationException);
+            } else if (value > MAX_SEARCH_RESULTS) {
```

Comment:
Should we make this validation a bit more consistent with [how we check result windows](https://github.com/elastic/elasticsearch/blob/8766e71d19502eda0bacabbb7f77c2b0d60b7197/server/src/main/java/org/elasticsearch/search/DefaultSearchContext.java#L206) in `_search`?

Reply:
Even though technically they are related, I'd like to keep the two separated. Synonyms and search should be separate concerns, and I think it could be confusing to the users if we relate the two (even though they are related underneath)

Reply:
I have the opposite perspective, I think that consistency in how our APIs behave makes things easier for users to understand, rather than knowing that the `_search` API and CRUD APIs have different behaviors. WDYT?

Reply:
Sorry there are two things here:
1. An error message, and I think we should have an error message different from `_search` request, as the error message of the search request suggests actions that a user can't do for synonyms.
2.  Checks for values. And @kderusso's point made me think that the current checks are indeed incorrect, and we should make these checks similar with `_search` request, meaning that the sum of `from+size` should not exceed `10_000`; otherwise we will get errors down the road from the search.

Reply:
Got it, I thought this was related to the error message and not on the actual validation. Thanks for clarifying!